convert2rhel/pkgmanager/handlers/yum/callback.py

```diff
@@ -0,0 +1,191 @@
+        :type frac: int
+        :param fread: String containing how much bytes was read in the download.
+        :type fread: str
+        :param ftime: String that contains the remaining time for te package to be finish the download.
+        :type ftime: str
+        """
+        # The base API will call this callback class on every package update,
+        # not matter if it is the same update or not, so, the below statement
+        # prevents the same message being sent more than once to the user.
+        if self.last_package_seen != name:
+            # Metadata download abut repositories will be sent to this class too.
+            if name.endswith(".rpm"):
+                loggerinst.info("Downloading package: %s", name)
+            else:
+                loggerinst.debug("Downloading repository metadata: %s", name)
+
+        self.last_package_seen = name
+
+
+class TransactionDisplayCallback(pkgmanager.TransactionDisplay, object):
+    """Transaction display callback for YUM transaction."""
+
+    def __init__(self):
+        """Constructor that overrides initialization for SimpleCliCallBack()."""
+        super(TransactionDisplayCallback, self).__init__()
+        # Hold the last package name to not print it twice, avoiding
+        # spamming msgs.
+        self.last_package_seen = None
+
+    def event(self, package, action, te_current, te_total, ts_current, ts_total):
+        """Process and output the RPM operations in the transaction.
+
+        .. note::
+            Yum API send the `package` paramter as two different types :holding-back-tears:.
+            If the package comes as a `str`, then it's a cleanup, if it's a
+            `yum.sqlitesack.YumAvailablePackageSqlite`, then it's a normal
+            package being processed.
+
+        :param package: Package being processed in the transaction.
+        :type param: str | yum.sqlitesack.YumAvailablePackageSqlite
+        :param action: The type of the action being used for the current
+            `package`. Only values that came from `rpmtrans.RPMBaseCallback.action` are valid.
+        :type action: int
+        :param te_current: How much work was already been done in the transaction.
+        :type te_current: int
+        :param te_total: How much work in total is present in the transaction.
+        :type te_total: int
+        :param ts_current: Number, in order, of the current transaction in the transaction set.
+        :type ts_current: int
+        :param ts_total: How much transactions are present in the transaction set.
+        :type ts_total: int
+        """
+        message = "%s: %s [%s/%s]"
+
+        # We convert the package here to a str because we just stand a
+        # normal-standard str comparision rather than what is implemented in
+        # `yum.sqlitesack.YumAvailablePackageSqlite`, as this class does
+        # version comparision and a bunch of other stuff. We don't care about
```

Comment:
```suggestion
        # Here we convert the package to a str because we just need to do a
        # standard str comparision rather than what is implemented in
        # `yum.sqlitesack.YumAvailablePackageSqlite` (that class does
        # version comparison and a bunch of other stuff). We don't care about
```